### DISCUSSION TITLE
How do I get Static Export to create directories for my html files?

### DISCUSSION THREAD

**Discussion by leo-petrucci** [vercel/next.js#15221] [Help]
I'm having trouble with Static Exporting.

When running in development mode, I can visit `example.com/contact-us`, and that will render the correct page. However, after exporting the page `example.com/contact-us` will return a 404 error.

This is caused by Next.js creating a `/contact-us.html` file rather than `/contact-us/index.html`.

I tried moving the contact-us page in `/pages/contact-us/index.js` but that has the same output. I can get Nextjs to create a folder if I name my Javascript file anything but `index` like so `/pages/contact-us/contact-us.js`, but that will then output `/contact-us/contact-us.html`.

How can I solve this?

**Comment by leo-petrucci** [comment]
I managed to get non-dynamic routes working:

```
// This maps `/pages/contact-us/index.js to /contact-us/index.html
module.exports = {
  exportPathMap: async function () {
    return {
      "/": { page: "/" },
      "/contact-us/index": { page: "/contact-us" },
    };
  },
}
```

But the same isn't working for dynamic routes:
```
// Trying to map /pages/info/[slug]/index.js 
  exportPathMap: async function () {
    return {
      "/": { page: "/" },
      "/contact-us/index": { page: "/contact-us" },
      "/info/privacy-notice/index": {
        page: "/info/[slug]",
        query: { slug: "privacy-notice" },
      },
    };
  },
```

I tried setting my route as `/pages/info/[slug].js` as well as `/pages/info/[slug]/index.js` but neither compiles correctly. I get the error:

```
Error occurred prerendering page "/info/privacy-notice/index". Read more: https://err.sh/next.js/prerender-error
Error: The provided export path '/info/privacy-notice/index' doesn't match the '/info/[slug]' page.
Read more: https://err.sh/vercel/next.js/export-path-mismatch
Exporting (5/5)
Error: Export encountered errors on following paths:
        /info/privacy-notice/index
```

Any idea how to get this working?

**Reply by leo-petrucci** [reply]
bump

**Reply by sajadgm** [reply]
Did you find the solution ?
